Source/Modules/javascript.cxx

```diff
@@ -2852,7 +2847,22 @@ int TypeScriptTypes::memberfunctionHandler(Node *n)
 {
   if (generateTsTypes)
   {
-    tsTypeInterfaceDeclaration->addMemberFunction(n);
+    String *functionName = Getattr(n, "sym:name");
+    String *functionPrototype = NewString(functionName);
+    Append(functionPrototype,"?(");
+    ParmList *l = Getattr(n, "parms");
+    Parm *p;
+    int i;
+    for (i = 0, p = l; p; i++) {
+      if ( i > 0) {
+        Append(functionPrototype,", ");
+      }
```

Comment:
This is a bit confusing as `i` is not really used as the loop index. I'd do it like this:
```suggestion
    ParmList* const l = Getattr(n, "parms");
    for (Perm* p = l; p; p = nextSibling(p)) {
      if ( p != l ) {
        Append(functionPrototype, ", ");
      }
```
(we could make `l` internal to the loop too, but this is probably not worth it -- but renaming it to `parms` might be).